python/smqtk/utils/configuration.py

```diff
@@ -60,10 +60,27 @@ def get_default_config(cls):
         turning those argument names into configuration dictionary keys. If any
         of those arguments have defaults, we will add those values into the
         configuration dictionary appropriately. The dictionary returned should
-        only contain JSON compliant value types.
+        only contain JSON compliant value types. If the default arguments are
+        not JSON compliant then the function should be overriden to convert
+        the convert the default to a JSON compliant stand in.
 
         It is not be guaranteed that the configuration dictionary returned
         from this method is valid for construction of an instance of this class.
+        >>> import math
+        >>> class NonCompliantDefault(Configurable):
+        ...     def __init__(self, power_func=math.pow):
+        ...         self.power_func=power_func
+        ...     @classmethod
+        ...     def get_default_config(cls):
+        ...         default = super(NonCompliantDefault, cls).get_default_config()
+        ...         default['power_func'] = {'module': 'math',
+        ...                                  'attribute': 'pow'}
+        ...         return default
+        >>> NonCompliantDefault.get_default_config()
+        {'power_func': {'module': 'math', 'attribute': 'pow'}}
```

Comment:
This fails on python 2.x unit tests due to differences in python versions' handling of dictionary internals and str-coersion. Since I don't think its appropriate yet to completely deprecate python 2 support, maybe a more specific assert is appropriate here:
```suggestion
        >>> NonCompliantDefault.get_default_config() == {'power_func': {'module': 'math', 'attribute': 'pow'}}
        True
```

Same for the next check as well.

Reply:
(probably do something sensible with line length here that I didn't format correctly).